### DISCUSSION TITLE
Implementing the functionality to run Actions of a Symbol widget by clicking on the widget

### DISCUSSION THREAD

**Discussion by abrahamwolk** [ControlSystemStudio/phoebus#2753] [General]
For Symbol widgets it is possible to specify Actions in the Display Editor. However, specified Actions are not run when clicking on a widget. How are the specified Actions supposed to be run?

I am considering implementing the functionality to run Actions of a Symbol widget by clicking on the widget. (Perhaps a checkbox would have to be enabled for the functionality to be activated.)

**Comment by kasemir** [comment]
See https://github.com/ControlSystemStudio/phoebus/wiki/Display-Builder-Compatibility#actions-action-buttons

**Reply by abrahamwolk** [reply]
Thanks for the link to the documentation!

It makes sense that it is obvious what is clickable and not. However, for some UIs it can be useful to be able to create custom graphics that are clickable, for instance "buttons" that are not rectangular, and that are part of a larger graphical representation. For cases like this, it would be useful to be able to (optionally) enable running the Actions of a Symbol widget by clicking on the widget (or more precisely, the non-transparent part of the Symbol) in question.

**Comment by kasemir** [comment]
The 'BOY' implementation had an option like "Run actions on click", don't remember exact name, on the _actions_.
That doesn't make complete sense because when you attach an action to an action button, then of course you want to run the action on click. For a text entry, it would be very odd to run the action on click because then how can you start editing the text?

So a "run actions on click" property on selected widgets, for example the symbol, makes more sense.
If you have nothing else to do, go ahead and add that.
